lib/src/css_parser.dart

```diff
@@ -105,15 +121,120 @@ class DeclarationVisitor extends css.Visitor {
+        left.width = expressionToBorderWidth(borderWidths.first);
+        right.width = expressionToBorderWidth(borderWidths.first);
+      }
+    }
+    if (borderStyles != null) {
+      top.style = expressionToBorderStyle(borderStyles.first);
+      if (borderStyles.length == 4) {
+        right.style = expressionToBorderStyle(borderStyles[1]);
+        bottom.style = expressionToBorderStyle(borderStyles[2]);
+        left.style = expressionToBorderStyle(borderStyles.last);
+      }
+      if (borderStyles.length == 3) {
+        left.style = expressionToBorderStyle(borderStyles[1]);
+        right.style = expressionToBorderStyle(borderStyles[1]);
+        bottom.style = expressionToBorderStyle(borderStyles.last);
+      }
+      if (borderStyles.length == 2) {
+        bottom.style = expressionToBorderStyle(borderStyles.first);
+        left.style = expressionToBorderStyle(borderStyles.last);
+        right.style = expressionToBorderStyle(borderStyles.last);
+      }
+      if (borderStyles.length == 1) {
+        bottom.style = expressionToBorderStyle(borderStyles.first);
+        left.style = expressionToBorderStyle(borderStyles.first);
+        right.style = expressionToBorderStyle(borderStyles.first);
+      }
+    }
+    if (borderColors != null) {
+      top.color = expressionToColor(borderColors.first);
+      if (borderColors.length == 4) {
+        right.color = expressionToColor(borderColors[1]);
+        bottom.color = expressionToColor(borderColors[2]);
+        left.color = expressionToColor(borderColors.last);
+      }
+      if (borderColors.length == 3) {
+        left.color = expressionToColor(borderColors[1]);
+        right.color = expressionToColor(borderColors[1]);
+        bottom.color = expressionToColor(borderColors.last);
+      }
+      if (borderColors.length == 2) {
+        bottom.color = expressionToColor(borderColors.first);
+        left.color = expressionToColor(borderColors.last);
+        right.color = expressionToColor(borderColors.last);
+      }
+      if (borderColors.length == 1) {
+        bottom.color = expressionToColor(borderColors.first);
+        left.color = expressionToColor(borderColors.first);
+        right.color = expressionToColor(borderColors.first);
+      }
+    }
+    return Border(
+        top: BorderSide(width: top.width, color: top.color, style: top.style),
+        right: BorderSide(width: right.width, color: right.color, style: right.style),
+        bottom: BorderSide(width: bottom.width, color: bottom.color, style: bottom.style),
+        left: BorderSide(width: left.width, color: left.color, style: left.style)
+    );
+  }
+
+  static double expressionToBorderWidth(css.LiteralTerm value) {
+    switch(value.text) {
+      case "thin":
+        return 2.0;
+      case "medium":
+        return 4.0;
+      case "thick":
+        return 6.0;
+      default:
+        return double.tryParse(value.text.replaceAll(new RegExp(r'\s+(\d+\.\d+)\s+'), ''));
```

Comment:
I think you are safe-guarding again non-digit characters here, but you are just stripping them... I think we should discard any input altogether instead if it is not a number. For example, we don't want the input `l33t` to parse as `33` I think, but just not 'work'.

Reply:
Hm this is tricky, I agree but at the same time the only other solution is to return `null`, in which case the border width becomes the default set by flutter. I would hope that people are inputting well-formed HTML, since obviously `border-width` only accepts numbers and those three terms. I did make this change, I'll leave it up to you to decide the best course of action.